lib/Sema/CSSolver.cpp

```diff
@@ -345,24 +345,60 @@ bool ConstraintSystem::simplify() {
     auto *constraint = &ActiveConstraints.front();
     deactivateConstraint(constraint);
 
+    if (isDebugMode()) {
+      auto &log = llvm::errs();
+      log.indent(solverState->getCurrentIndent());
+      log << "(considering -> ";
+      constraint->print(log, &getASTContext().SourceMgr);
+      log << "\n";
+
+      if (constraint->getKind() != ConstraintKind::Disjunction &&
```

Comment:
Nit: please add a comment about why this is not printed for disjunctions and conjunctions.